Task_DEV_4/Task_DEV_4/Parallelogram.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+namespace Task_DEV_4
+{
+    class Parallelogram : Square
+    {
+        public double Height { get; set; }
+        public double Angle { get; set; }
+        public double Width { get; set; }
+
+        public Parallelogram(double length, double height) : base(length)
+        {
+            Height = height;
+        }
+
+        public Parallelogram(double length, double angle, double width) : base(length)
+        {
+            Width = width;
+            Angle = angle;
+        }
+
+        public override double CalculateArea()
+        {
+            if (Height > 0 && Length > 0)
+            {
+                return Length * Height;
+            }
+            else if (Length > 0 && Width > 0 && Angle > 0)
+            {
+                return Length * Width * Math.Sin(Angle);
+            }
+            else
+            {
+                throw new ArgumentOutOfRangeException();
```

Comment:
It would be more prefferable to provide your message to the constructor of exception smth like:
`throw new ArgumentOutOfRangeException("Your message");`
Then You can do in catch block smth like this
```
catch (ArgumentOutOfRangeException e)
{
    Console.WriteLine(e.Message");
}
```

Reply:
Done.